submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/security/web/SecurityResource.kt

```diff
@@ -45,6 +46,15 @@ class SecurityResource(
         securityService.registerUser(register)
     }
 
+    @PostMapping(value = ["/check-user"])
```

Comment:
I think this name for this particular endpoint is very misleading. We should go for something like `check-register` or `search-register`; something that indicates that this endpoint actually performs two operations.

Said this, I'm not sure about how REST compliant this is, maybe we need just to have a check endpoint and let them decide whether or not to use the already existing register endpoint.

Besides, the actual registration endpoint has a lot of info that this one doesn't like the password, whether or not the notifications are enabled, etc. How will this be handled later for the user? I'm just worried this could become a workaround for our actual registration process and we should be careful now that our API is having external users.

![image](https://user-images.githubusercontent.com/19352419/102331064-3d334280-3f82-11eb-8203-8cebd2de057a.png)

Reply:
hahaha , the issue with the registration one is that is used for humans this is only for machine integration that need to register users. As array express or PMC so as we do not want to for example allow then to provide i.e password or other information that they do not have this is the endpoint for that :).


I agree that `check-register`is a much better name

Reply:
Got it